### PR TITLE
ipatests: Test ldapsearch with base scope works for compat tree.

### PR DESCRIPTION
**Pull request by amore17**
Related: https://bugzilla.redhat.com/show_bug.cgi?id=1958909

### CODE REVIEW

File: ipatests/test_integration/test_adtrust_install.py [hunk 1]
```diff
@@ -276,6 +278,21 @@ def test_ipa_user_pac(self):
...
+        """Test ldapsearch against compat tree is working
+
+        This to ensure that ldapsearch with base scope is not failing.
+
+        related: https://bugzilla.redhat.com/show_bug.cgi?id=1958909
+        """
+        tasks.kinit_admin(self.master)
+        base_dn = str(self.master.domain.basedn)
+        base = "cn=admins,cn=groups,cn=compat,{basedn}".format(basedn=base_dn)
+        tasks.ldapsearch_dm(self.master, base, ldap_args=[], scope='base')
```
Comment (mrizwan93) [inline ipatests/test_integration/test_adtrust_install.py:294]: could you also make ldap search with scope as sub

File: ipatests/test_integration/test_adtrust_install.py [hunk 2]
```diff
@@ -276,6 +278,21 @@ def test_ipa_user_pac(self):
             tasks.kinit_admin(self.master)
             self.master.run_command(['ipa', 'user-del', user])
 
+    @pytest.mark.xfail(
+        osinfo.id == 'fedora',
+        reason='Test requires slapi-nis with fix', strict=True)
```
Comment (mrizwan93) [inline ipatests/test_integration/test_adtrust_install.py:283]: would be good if required slapi-nis version is mentioned OR you can check for slapi-nis version in the test and skip/xfail depending upon version.

**Review by mrizwan93 — COMMENTED** [review]
Hello @amore17 Thakns for the patch.

Could you please update commit message elaborating more about test. Also some inline comments.

**Comment by amore17** [comment]
Tested on RHEL with fixed slapi-nis : [report](http://idm-artifacts.usersys.redhat.com/amore/trigger//1366/test-suite/report.html.gz)

File: ipatests/test_integration/test_adtrust_install.py (hunk 2 above)
Comment (amore17) [reply ipatests/test_integration/test_adtrust_install.py]: Added check for version and marked as xfail.

**Comment by amore17** [comment]
@mrizwan93  Tested with RHEL using fixed slapi-nis

**Review by flo-renaud — COMMENTED** [review]
Hi @amore17 
thanks for the PR. I have inline comments + a general suggestion. The compat plugin can be tested without any AD trust and the test could be moved to test_commands.py for instance, where only a single machine is needed.

**Comment by amore17** [comment]
Tested this on RHEL9 with slapi-nis-0.56.7-2.el9.x86_64

File: ipatests/test_integration/test_adtrust_install.py [hunk 3]
```diff
@@ -276,6 +279,25 @@ def test_ipa_user_pac(self):
...
+        This to ensure that ldapsearch with base scope is not failing.
+
+        related: https://bugzilla.redhat.com/show_bug.cgi?id=1958909
+        """
+        version = self.master.run_command(
+            ["rpm", "-qa", "--qf", "%{VERSION}", "slapi-nis"]
+        )
+        if osinfo.id == 'fedora' and parse_version(
+                version.stdout_text) <= parse_version("0.56.7"):
+            pytest.skip("Test requires slapi-nis with fix")
```
[the remark this answers is not in the document]
Comment (amore17) [reply ipatests/test_integration/test_adtrust_install.py]: Its  only for fedora. On RHEL patched version are 0.56.7-2 and 0.56.6-3.

File: ipatests/test_integration/test_adtrust_install.py [hunk 4]
```diff
@@ -276,6 +279,25 @@ def test_ipa_user_pac(self):
...
+        """Test ldapsearch against compat tree is working
+
+        This to ensure that ldapsearch with base scope is not failing.
+
+        related: https://bugzilla.redhat.com/show_bug.cgi?id=1958909
+        """
+        version = self.master.run_command(
+            ["rpm", "-qa", "--qf", "%{VERSION}", "slapi-nis"]
+        )
+        if osinfo.id == 'fedora' and parse_version(
```
[the remark this answers is not in the document]
Comment (amore17) [reply ipatests/test_integration/test_adtrust_install.py]: updated.

**Comment by flo-renaud** [comment]
Thanks for the path, LGTM.

**Comment by flo-renaud** [comment]
master:

* 7cac63faefc4a908b6bcf11dfe430ec4e7e8af8a ipatests: Test ldapsearch with base scope works with compat tree.
* cd665e2d5cfedc5f1f7c7d4ea180cf51af9b51af ipatests: skip test_basesearch_compat_tree on fedora.

**Pull request closed without merging**